### ISSUE TITLE
Save and Load Networks for each training run

### ISSUE THREAD

**Issue by mbecker12** [mbecker12/surface-rl-decoder#47]
After each training run, the trained model should be saved.

Additionally, there should be the option to load previous models on demand, to improve an existing model or to evaluate an existing model

**Comment by mbecker12** [comment]
Saving and loading works locally.

Need to test on cluster.
- [ ] saving
- [ ] loading

**Comment by mbecker12** [comment]
loading: need to move optimizer to device in learner process

**Comment by mbecker12** [comment]
saving seems to work on cluster

**Comment by mbecker12** [comment]
For final, thorough testing, this issue is blocked by #57

**Comment by mbecker12** [comment]
In the meantime, tested gather function in learner_util.py, 
```
policy_output_gathered = policy_output.gather(1, batch_action_indices)

    for i, action_idx in enumerate(batch_action_indices):
        assert policy_output_gathered[i, 0] == policy_output[i][action_idx]
```

And the max functionality a few lines below:

```
    target_output = _target_output.max(1)[0].detach()
    for i in range(batch_size):
        assert target_output[i] == max(_target_output[i])
```

**Comment by mbecker12** [comment]
For testing purposes, the image was run on a private azure VM instance. There, saving and loading, as well as GPU access seems to work fine (Tesla K80).
Also, deploying one actor on CPU and one on CUDA seemed to work fine

**Comment by mbecker12** [comment]
Make sure that the optimizer properly trains the model it is assigned to; even after loading model + optimizer and transferring them to their devices

**Comment by mbecker12** [comment]
Slurm job 57239 prints network parameters. They are being trained on GPU.
Also, merged successfully #58
